addon/build-waiter.ts

```diff
@@ -1,14 +1,194 @@
+import { Primitive, TestWaiter, TestWaiterDebugInfo, WaiterName } from 'ember-test-waiters';
+
 import { DEBUG } from '@glimmer/env';
-import { TestWaiter, ITestWaiter } from 'ember-test-waiters';
-import NoopTestWaiter from './noop-test-waiter';
+import Token from './token';
+import { register } from './waiter-manager';
+
+function getNextToken(): Token {
+  return new Token();
+}
+
+/**
+ * A class providing creation, registration and async waiting functionality.
+ *
+ * @public
```

Comment:
I think this should be private

```suggestion
 * @private
```